src/Controller/Pia/ProcessingController.php

```diff
@@ -73,4 +74,145 @@ public function listAction(Request $request)
 
         return $this->view($collection, Response::HTTP_OK);
     }
+
+    /**
+     * Shows one Processing by its ID.
+     *
+     * @Swg\Tag(name="Processing")
+     *
+     * @FOSRest\Get("/processings/{id}", requirements={"id"="\d+"})
+     *
+     * @Swg\Response(
+     *     response=200,
+     *     description="Returns one Processing",
+     *     @Swg\Schema(
+     *         type="object",
+     *         ref=@Nelmio\Model(type=Processing::class, groups={"Default"})
+     *     )
+     * )
+     *
+     * @Security("is_granted('CAN_SHOW_PROCESSING')")
+     *
+     * @return View
+     */
+    public function showAction(Request $request, $id)
+    {
+        return $this->showEntity($id);
+    }
+
+
+    /**
+     * Creates a Processing.
+     *
+     * @Swg\Tag(name="Processing")
+     *
+     * @FOSRest\Post("/processings")
+     *
+     * @Swg\Response(
+     *     response=200,
+     *     description="Returns the newly created Processing",
+     *     @Swg\Schema(
+     *         type="object",
+     *         ref=@Nelmio\Model(type=Processing::class, groups={"Default"})
+     *     )
+     * )
+     *
+     * @Security("is_granted('CAN_CREATE_PROCESSING')")
+     *
+     * @return array
+     */
+    public function createAction(Request $request)
+    {
+        
+        $folder = $this->getResource($request->get('folder_id', -1), Folder::class);
+        $entity = $this->get('jms_serializer')->deserialize($request->getContent(), $this->getEntityClass(), 'json');
```

Comment:
Why unserializing the whole entity ?

Fetching only necessary fields `$request->get('name', null)` is not sufficient ?

Also please note that using the container will be deprecated soon (you should inject the service `jms_serializer` in the controller itself with type `JMS\Serializer\SerializerInterface`).

Reply:
You're right. I copied the method from RestController but it's not relevant here.